cli/azd/pkg/project/framework_service_dotnet_container_publish.go

```diff
@@ -0,0 +1,83 @@
+package project
+
+import (
+	"context"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/async"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+)
+
+type dotNetContainerPublishProject struct {
+	inner FrameworkService
+}
+
+func NewDotNetContianerPublishProject() CompositeFrameworkService {
```

Comment:
Not sure if this is better than peppering the existing Docker target with some `if`'s, but it felt like maybe it was playing a bit more by the conventions.  Not wed to the idea of introducing a new `CompositeFrameworkService` and instead just adding more logic to the docker one, if folks think that would be helpful.